ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -19,6 +20,66 @@
 #include "ttnn/tensor/host_buffer/types.hpp"
 #include "ttnn/cpp/ttnn/tensor/enum_types.hpp"
 
+namespace ttnn {
+    /**
+SimpleShape is a temporary measure aimed at making a clear distinction between Shape/LegacyShape when padding information is stripped.
+Context:
+    Both Shape and LegacyShape can carry padding information.
+    And we use Shape or LegacyShape to carry full shape info or separately logical or physical shape.
+    Absence of distinction between full shape and logical shape leads to confusion and makes further refactoring harder.
+Plan:
+    We want to replace `Shape Shape::with_tile_padding() const` with `SimpleShape padded_shape() const`
+    We will clearly see where full shape is used vs logical or physical shape is used.
+    Need to split .hpp and .cpp
+**/
+class SimpleShape {
+public:
+    template <typename T>
+    explicit SimpleShape(T&& shape) : value{std::forward<T>(shape)} {}
+
+    template<std::size_t N>
+    bool operator==(const std::array<uint32_t, N> &other) const {
+        bool sameSize = value.size() == N;
+        return sameSize && std::equal(value.begin(), value.end(), other.begin());
+    }
+
+    bool operator==(const SimpleShape &other) const;
+    bool operator==(const std::vector<uint32_t> &other) const;
+
+    uint32_t operator[](int32_t index) const;
+    uint32_t &operator[](int32_t index);
+
+    size_t rank() const { return this->value.size(); }
+
+    uint64_t volume() const;
```

Comment:
@ayerofieiev-tt Given your comment "These two calls are identical, except its better to ask tensor directly", would you want to make volume() protected or private with Tensor or it's calls as friend(s)?

Reply:
It was confusing for LegacyShape and Shape because they contained both Shape/Padded Shape and Padding info.
It was not clear if volume was supposed to be a Padded-based volume or unpadded-based volume.
Thats not a problem for SimpleShape.